### PR TITLE
add --tray option to start without window visible

### PR DESCRIPTION
**Pull request by pfps**
So 
solaar
will start with the window visible (this is a change)
and
solaar -tray
will start with only the icon in the tray.

Documentation has not yet been updated - waiting for my other documentation changes to be pulled so that I don't have to merge changes.  
Update: Documentation update may not be necessary --help provides information

### CODE REVIEW

**Comment by pfps** [comment]
For issue https://github.com/pwr-Solaar/Solaar/issues/593

**Comment by FFY00** [comment]
Thanks, it all looks good! I just want to test the patch before merging. We should also ship .desktop files with the app. If you want to add that I would appreciate it as I lack time right now.

**Comment by pfps** [comment]
I think that the .desktop file is already there - share/applications/solaar.desktop
It is even installed in setup.py

**Comment by FFY00** [comment]
We should add one for the tray them.

**Comment by pfps** [comment]
So a separate desktop file for starting up with the window open and one for starting up closed?  Isn't that a bit overkill?  I'm not sure that having both in your application menu is best.  However, a separate desktop file that isn't put into menus sounds good.

**Comment by pfps** [comment]
OK, a new autostart file (in a subdirectory so that it can have the same name) that starts without a window and is put in the autostart directory.

**Comment by pfps** [comment]
Grr - this ended up with some commits that are already in the master branch.  Hopefully that is OK.

**Comment by FFY00** [comment]
Can you use rebase to sync with master instead of merge. I think to fix the current history you need to do `git reset --hard HEAD~1` and then `git rebase master`.

**Comment by pfps** [comment]
I think I screwed up so bad that I will have to reconstruct using patches.  :-(
This may require putting in new pull requests.

**Comment by FFY00** [comment]
Nah, just force push your new history to the remote repo. Do you need any help?

**Comment by pfps** [comment]
OK, that worked.

File: lib/solaar/ui/window.py [hunk 1]
```diff
@@ -737,6 +737,7 @@ def init():
 	_info = _create_info_panel()
 	_empty = _create_empty_panel()
 	_window = _create()
+	if (not tray_only) : _window.present()
```
Comment (Lekensteyn) [inline lib/solaar/ui/window.py]: Style nitpick:
```python
if not tray_only:
    _window.present()
```

**Comment by Lekensteyn** [comment]
Would it make sense to add a preference for the tray icon setting instead of, or in addition to the command line flag? Earlier on there was a feature request to hide the tray icon (#84, unmerged PR in #161). That should remove the need for a dedicated .desktop file. Possible settings:

Startup behavior:
- Show both the window and tray icon.
- Show tray icon only without window.
- Show window only without tray icon.

**Comment by pfps** [comment]
That would be possible, but some care would have to be taken to change to the right behaviour.
I think that a better way of describing those options would be:
1. Run without any tray involvement at all - implies window visible at startup and closing window quits the program (default)
2. Run with tray icon - implies closing window does not quit the program. (--tray)
2a. Start with window visible (--window=visible - default)
2b. Start with window not visible (--window=invisible)
This would be a larger change when running with no flag though.

The different behaviour on window close worries me a bit.  The other programs I know of that use the tray in a long-term fashion (quodlibet, power manager, network manager, bluetooth manager, clipboard) always run with the tray on.  So window close doesn't ever cause a quit.

File: lib/solaar/ui/window.py (hunk 1 above)
Comment (pfps) [reply lib/solaar/ui/window.py]: Style buffed as suggested

**Comment by FFY00** [comment]
@pfps ping me when this is ready.

**Comment by pfps** [comment]
@FFY00  I'll fix up the commits etc. and ping you when done.

**Comment by pfps** [comment]
@FFY00  OK, this pull should be ready to go.
I redid some of the commits and changed their names.  This required a force push.

**Comment by pfps** [comment]
@FFY00 The tray stuff has been working well for me.  However, I did fiddle with the install script and that should be tested, as I only have one machine that I can do an install on.

**Comment by FFY00** [comment]
Okay

**Comment by pfps** [comment]
Addresses #552

**Comment by Lekensteyn** [comment]
Thanks!

**Comment by alvarobrey** [comment]
What happened to this? I don't see the option in `solaar --help` in 1.0.3

**Pull request merged**